RecoCTPPS/TotemRPLocal/src/CTPPSDiamondRecHitProducerAlgorithm.cc

```diff
@@ -11,42 +11,51 @@
 //----------------------------------------------------------------------------------------------------
 
 CTPPSDiamondRecHitProducerAlgorithm::CTPPSDiamondRecHitProducerAlgorithm( const edm::ParameterSet& iConfig ) :
-  ts_to_ns_( iConfig.getParameter<double>( "timeSliceNs" ) ),
-  t_shift_( iConfig.getParameter<int>( "timeShift" ) )
+  ts_to_ns_( iConfig.getParameter<double>( "timeSliceNs" ) )
 {}
 
 void
 CTPPSDiamondRecHitProducerAlgorithm::build( const CTPPSGeometry* geom, const edm::DetSetVector<CTPPSDiamondDigi>& input, edm::DetSetVector<CTPPSDiamondRecHit>& output )
 {
-  for ( edm::DetSetVector<CTPPSDiamondDigi>::const_iterator vec = input.begin(); vec != input.end(); ++vec ) {
-    const CTPPSDiamondDetId detid( vec->detId() );
+  for ( const auto& vec : input ) {
+    const CTPPSDiamondDetId detid( vec.detId() );
 
-    if ( detid.channel() > 20 ) continue;              // VFAT-like information, to be ignored by CTPPSDiamondRecHitProducer
+    if ( detid.channel() > 20 ) continue; // VFAT-like information, to be ignored
 
+    // retrieve the geometry element associated to this DetID
     const DetGeomDesc* det = geom->getSensor( detid );
+
+    if ( det->parents().empty() )
+      edm::LogWarning("CTPPSDiamondRecHitProducerAlgorithm") << "The geometry element for " << detid << " has no parents. Check the geometry hierarchy!";
```

Comment:
If  ```det->parents().empty()```  the z_pos computed at L35 will make the execution crash.
I would make this a LogError, and then you can assert either on ```!det->parents().empty()``` , or (if you want to continue the process anyhow) define a default/dummy z_pos in that case